src/SDKs/Search/DataPlane/Search.Tests/Tests/SkillsetsTests.cs

```diff
@@ -0,0 +1,610 @@
+                    searchClient.Skillsets.CreateOrUpdateWithHttpMessagesAsync(skillset.Name, skillset).Result;
+                Assert.Equal(HttpStatusCode.Created, response.Response.StatusCode);
+            });
+        }
+
+        [Fact]
+        public void CreateOrUpdateUpdatesWhenSkillsetExists()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+
+                Skillset skillset = CreateTestOcrSkillset(1);
+
+                AzureOperationResponse<Skillset> response =
+                    searchClient.Skillsets.CreateOrUpdateWithHttpMessagesAsync(skillset.Name, skillset).Result;
+                Assert.Equal(HttpStatusCode.Created, response.Response.StatusCode);
+
+                skillset = CreateTestOcrSkillset(2, skillset.Name);
+                response =
+                    searchClient.Skillsets.CreateOrUpdateWithHttpMessagesAsync(skillset.Name, skillset).Result;
+                Assert.Equal(HttpStatusCode.OK, response.Response.StatusCode);
+            });
+        }
+
+        [Fact]
+        public void GetOcrSkillsetReturnsCorrectDefinition()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+                CreateAndGetSkillset(searchClient, CreateTestOcrSkillset(1));
+            });
+        }
+
+        [Fact]
+        public void GetOcrSkillsetWithShouldDetectOrientationReturnsCorrectDefinition()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+                CreateAndGetSkillset(searchClient, CreateTestOcrSkillset(repeat: 1, shouldDetectOrientation: true));
+            });
+        }
+
+        [Fact]
+        public void DeleteSkillsetIsIdempotent()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+
+                Skillset skillset = CreateTestOcrSkillset(1);
+
+                // Try delete before the Skillset even exists.
+                AzureOperationResponse deleteResponse =
+                    searchClient.Skillsets.DeleteWithHttpMessagesAsync(skillset.Name).Result;
+                Assert.Equal(HttpStatusCode.NotFound, deleteResponse.Response.StatusCode);
+
+                searchClient.Skillsets.Create(skillset);
+
+                // Now delete twice.
+                deleteResponse = searchClient.Skillsets.DeleteWithHttpMessagesAsync(skillset.Name).Result;
+                Assert.Equal(HttpStatusCode.NoContent, deleteResponse.Response.StatusCode);
+
+                deleteResponse = searchClient.Skillsets.DeleteWithHttpMessagesAsync(skillset.Name).Result;
+                Assert.Equal(HttpStatusCode.NotFound, deleteResponse.Response.StatusCode);
+            });
+        }
+
+        private void CreateAndGetSkillset(SearchServiceClient searchClient, Skillset expectedSkillset)
```

Comment:
There's quite a bit of missing test coverage here. You should test:
- Every enum value
- Every default value (don't set any optional properties when creating skills, then make sure the defaults that come back match your expectations)
- That Get throws when the given skillset name doesn't exist
- That ETags/AccessCondition works the way you expect
- That you can actually execute a skillset (I don't see how to do that with the code you have so far)

For examples of many of these, see the `IndexManagementTests`: https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Search/DataPlane/Search.Tests/Tests/IndexManagementTests.cs

Reply:
Thanks. I will add more tests for coverage. We cannot execute the skillset by the SkillsetOperations. The Skillset needs to be executed with Indexer. I checked the IndexerOperations and I think I missed to add the SkillsetName into Indexer. I will do that and add some tests to execute Indexer with Skillset scenario. Thanks.

Reply:
I think Indexer should also have output field mappings now. @chaosrealm correct me if I'm wrong...